scipy/integrate/_quadrature.py

```diff
@@ -1035,3 +1038,131 @@ def newton_cotes(rn, equal=0):
+
+    n_points_int = np.int64(n_points)
+    if n_points != n_points_int:
+        message = "`n_points` must be an integer."
+        raise TypeError(message)
+
+    if qmc_engine is None:
+        qmc_engine = stats.qmc.Halton(dim)
+    elif not isinstance(qmc_engine, stats.qmc.QMCEngine):
+        message = "`qrng` must be an instance of scipy.stats.qmc.QMCEngine."
+        raise TypeError(message)
+
+    return vfunc, ranges, n_points_int, qmc_engine, args, stats
+
+
+QMCQuadResult = namedtuple('QMCQuadResult', ['integral', 'standard_error'])
+
+
+def qmc_quad(func, ranges, *, n_points=1024, qrng=None, args=None):
+    """
+    Compute an integral in N-dimensions using Quasi-Monte Carlo quadrature.
+
+    Parameters
+    ----------
+    func : callable
+        The function to be integrated. Must accept separate arguments
+        ``x0, ..., xn`` corresponding with the coordinate in each dimension.
+        May also accept additional positional arguments specified with
+        `args`.
+    ranges : array-like
+        A two-dimensional array specifying the lower and upper integration
+        limits of each variable.
+    n_points : int, default: 1024
+        The number of Quasi-Monte Carlo points at which to evaluate the
+        integrand.
+    qrng : stats.qmc.QMCEngine, optional
+        An instance of the QMCEngine from which to sample QMC points.
+        The QMCEngine must be initialized to a number of dimensions
+        corresponding with the number of variables ``x0, ..., xn`` passed to
+        `func`. If a QMCEngine is not provided, `scipy.stats.qmc.Halton` will
+        be initialized with the number of dimensions determine from `ranges`
+        and the default `seed`.
+    args : sequence, optional
+        Extra arguments to pass to `func`. All extra arguments must be packed
+        into a sequence; the will be unpacked to be passed to `func`, e.g.
+        ``func(x1, x2, *args)``.
+
+    Returns
+    -------
+    result : object
+        The a result object with attributes:
+
+        integral : float
+            The estimate of the integral
+        standard_error :
+            The standard error of the
+
+    """
+    args = _qmc_quad_iv(func, ranges, n_points, qrng, args)
+    func, ranges, n_points, qrng, args, stats = args
+
+    sample = qrng.random(n_points)
+    # Can't check this in input validation or that would burn sampled
+    # points, which would be a problem for Sobol
+    if sample.shape[-1] != ranges.shape[0]:
+        message = ("`qrng` must be initialized with dimensionality equal to "
+                   "the number of variables in `ranges`, i.e., "
+                   "`qrng.random().shape[-1]` must equal `ranges.shape[0]`.")
+        raise ValueError(message)
+
+    x = stats.qmc.scale(sample, *ranges.T)
+    lb, ub = ranges.T
+    A = np.prod(ub - lb)
+    integrands = func(*x.T, *args)
+    integral = np.mean(integrands * A)
+    standard_error = stats.sem(integrands) * A
```

Comment:
I don't think this has any real relationship to the numerical error of the integral estimate. The `integrands` aren't i.i.d. samples around a mean value. The technique that [I have seen](http://www.math.wsu.edu/faculty/genz/papers/mvn.pdf) repeats the QMC ~10 times or so with different pseudorandom offsets in the (wrapped) unit hypercube, then computes the mean and standard deviation of those replicates.

Reply:
> The technique that I have seen repeats the QMC ~10 times

Yes, I thought about doing something like that, or at least taking the mean of subsets of points and comparing them (e.g. jacknife or some batch jacknife type of thing). Then I thought this would be just a little simpler to code, and enough to get the conversation started. It was a last minute addition.

> The integrands aren't i.i.d. samples around a mean value

If these were regular MC points, at least, I think there would be some justification for this, right?

- There is some probability distribution underlying the elements in `integrands` for which the elements _are_ i.i.d. samples.
- Sample means of i.i.d. samples from this distribution are approximately normally distributed about the population mean, assuming the CLT applies.

If that much is true, the s.e.m. can be used to generate a confidence interval for the mean, so it seemed like a reasonable estimate of the error _if these were regular MC points_. Anyway, I tested it with regular MC points, and the 1-sigma confidence intervals suggested by the `sem` are accurate - they contain the true value ~68% of the time.

It is definitely sloppy to treat the QMC points as though they are regular MC points, but I thought that the QMC error would tend to be _lower_ for well-behaved integrands, so this would be a reasonable error estimate for poorly-behaved integrands.

But again, I didn't look into the theory or other implementations or anything else. Just wanted to start the conversation. We could certainly do something more sophisticated.

Reply:
[Better reference.](https://core.ac.uk/download/pdf/82457836.pdf) The above paper still refers to an MC scheme for computing multivariate normal integrals, but [later versions](https://www.math.wsu.edu/faculty/genz/software/matlab/qsilatmvnv.m) use QMC with error estimates as I described.

Reply:
I'll take a look! Lmk which part of the argument was wrong (besides treating QMC points as regular MC points).

Reply:
Unless if you tautologically define "well-behaved integrand" as "one in which the QMC error is less than the MC error", there definitely are cases where the QMC variance is `n_points` times the MC variance. You do have to be particularly unlucky in your choice of integrand and QMC points, but it's hard to tell _a priori_ when this MC-appropriate method will fail to give a good estimate for the QMC error. [Our documentation](https://docs.scipy.org/doc/scipy/reference/stats.qmc.html) has more references (see the paragraphs on "Randomized QMC").

Reply:
OK, so mostly it's the treatment of QMC points as MC points that makes sense, and the assumption that QMC error would typically be lower than MC error. Makes sense.

Reply:
But our documentation also explicitly states:
![image](https://user-images.githubusercontent.com/6570539/187306258-76f99ffe-2891-4a67-a336-00a81a38530d.png)

This is what I'm doing here. Does the scrambling help us here @tupui?

Reply:
In this paragraph, the $R$ independent replications is the ~10 number that I referred to, not `n_points`.

Reply:
I see.  $n$ is a different symbol than $R$ for a reason : )

Reply:
RQMC refers to scrambling sequences yes. And yes R is the number of replicates or number of independent QMC samples. This is what I typically do, generate R different samples (from a fresh generator if Sobol' due to the mess with the power of 2 rule. Good thing here that you are using Halton as a default) and get the mean/var of the means/vars.

I am not sure you can do the MC CLT. (Well should also work since QMC samples are MC samples too). I have asked some friends about what we can do in practice (we can use the discrepancy to have some bounding on the errors, but I've never done it in practice and I am not sure how it really works).

Reply:
Hopefully I understood the [better reference](https://core.ac.uk/download/pdf/82457836.pdf) and fixed this!

Reply:
![image](https://user-images.githubusercontent.com/6570539/187621603-37c1e3b9-fb6e-4522-b437-94c7d84c60f5.png)

IIUC $q$ here is what we've been calling $R$, and for $\Delta_j$ I've used PRNG samples from the multivariate uniform distribution wrapped to the unit hypercube.

Reply:
I'll go ahead and resolve this, since it has been changed as discussed.